File: tools/src/tasks/io.py
import json
from pathlib import Path

LIB_DIR = Path(__file__).parent.resolve()
REPO_DIR = LIB_DIR.parent.parent.resolve()
OPENFGA_AUTH_MODEL_DIR = f"{REPO_DIR}/openfga"
OPENFGA_AUTH_JSON_FILE = f"{OPENFGA_AUTH_MODEL_DIR}/model.json"


def read_file(path: str) -> str:
    with open(path, "r") as file:
        data = file.read()
    return data


def read_json(path: str) -> dict:
    return json.loads(read_file(path))

File: src/lib/tasks/io.ts
import fs from "fs";
import path from "path";
import { dirname } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const REPO_DIR = path.join(__dirname, "../../../");
const AUTH_MODEL_DSL_FILE = path.join(REPO_DIR, "openfga/model.fga");
const AUTH_MODEL_JSON_FILE = path.join(REPO_DIR, "openfga/model.json");

function readText(path: string) {
  return fs.readFileSync(path).toString();
}

function readJSON(path: string) {
  return JSON.parse(readText(path));
}

function write(path: string, data: string) {
  fs.writeFileSync(path, data);
}

export function readAuthModelJSON() {
  return readJSON(AUTH_MODEL_JSON_FILE);
}

export function readAuthModelDSL() {
  return readText(AUTH_MODEL_DSL_FILE);
}

export function writeAuthModelJSON(data: string) {
  write(AUTH_MODEL_JSON_FILE, JSON.stringify(JSON.parse(data), null, 2));
}
